### PR TITLE
Add `array_cat()`, `array_append()`, `array_prepend()` and `||` 

### LINKED ISSUE #1107: Document `array_cat()`, `array_append()`, `array_prepend()` and `||`
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/5060

### Which part(s) of the docs might be affected or should be updated? And how?

https://www.risingwave.dev/docs/*/sql-function-array/

[Available](https://github.com/risingwavelabs/risingwave/commit/52b259ab19df7b750a7153c174ac132f9616a852) since v0.1.13, but may be okay if we leave the old versions not updated.

### Reference

https://www.postgresql.org/docs/15/functions-array.html

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

 Doc `array_cat()`, `array_append()`, `array_prepend()` and `||` under array functions.

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/5060

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1107

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-array.md [hunk 1]
```diff
@@ -4,16 +4,38 @@ slug: /sql-function-array
...
+array_append ( array, any_compatible ) → array
+```
+
+```sql title=Example 
+array_append ( array[66], array[123] ) → {66,233}
+```
+
+---
+
+### `array_cat`
```
Comment (stdrc) [inline docs/sql/functions-operators/sql-function-array.md]: trailing space

File: docs/sql/functions-operators/sql-function-array.md [hunk 2]
```diff
@@ -4,16 +4,38 @@ slug: /sql-function-array
...
 
-Returns the dimensions of *array* as a string. *array* must be one dimensional.
+Appends *any_compatible* to the end of the input array.
 
 ```sql title=Syntax
-array_dims ( array ) → string
+array_append ( array, any_compatible ) → array
+```
+
+```sql title=Example
```
Comment (stdrc) [inline docs/sql/functions-operators/sql-function-array.md]: Please remove the trailing space🤣

File: docs/sql/functions-operators/sql-function-array.md [hunk 3]
```diff
@@ -4,16 +4,38 @@ slug: /sql-function-array
...
 
 ```sql title=Example
-array_dims ( array[2,3,4] ) → [1:3]
+array_cat ( array[66], array[123] ) → {66, 123}
+
+array[66] || array[123] → {66, 123}
+
+array_cat(array[array[66]], array[233]) → {{66},{233}}
+
+array_cat(array[233], array[array[66]]) → {{233},{66}}
```
Comment (stdrc) [inline docs/sql/functions-operators/sql-function-array.md]: Please keep spacing consistent over the whole page.

Either:

```suggestion
array_cat ( array[66], array[123] ) → {66, 123}

array[66] || array[123] → {66, 123}

array_cat ( array[array[66]], array[233] ) → {{66}, {233}}

array_cat ( array[233], array[array[66]] ) → {{233}, {66}}
```

Or:

```suggestion
array_cat(array[66], array[123]) → {66, 123}

array[66] || array[123] → {66, 123}

array_cat(array[array[66]], array[233]) → {{66}, {233}}

array_cat(array[233], array[array[66]]) → {{233}, {66}}
```

File: docs/sql/functions-operators/sql-function-array.md [hunk 4]
```diff
@@ -4,16 +4,38 @@ slug: /sql-function-array
...
-Returns the dimensions of *array* as a string. *array* must be one dimensional.
+Appends *any_compatible* to the end of the input array.
 
 ```sql title=Syntax
-array_dims ( array ) → string
+array_append ( array, any_compatible ) → array
+```
+
+```sql title=Example 
+array_append ( array[66], array[123] ) → {66,233}
```
Comment (stdrc) [inline docs/sql/functions-operators/sql-function-array.md]: Either adding or not adding spaces between each pair of elements in array representation is OK, but should keep consistency.

File: docs/sql/functions-operators/sql-function-array.md [hunk 5]
```diff
@@ -4,16 +4,38 @@ slug: /sql-function-array
...
+---
+
+### `array_cat` 
+
+Concatenates two arrays with the same data type. The `||` operator can also be used.
+
+If the one of the input arrays is a 2-dimensional array, the other array will be appended within the first array as the last element if it is the second argument. The other array will be prepended within the first array as the first element if it is the first argument.
+
+```sql title=Syntax
+array_cat ( array,  array ) → array
```
Comment (stdrc) [inline docs/sql/functions-operators/sql-function-array.md]: Redundant space between the two arguments

```suggestion
array_cat ( array, array ) → array
```

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks.

File: docs/sql/functions-operators/sql-function-array.md [hunk 6]
```diff
@@ -4,16 +4,38 @@ slug: /sql-function-array
...
+Appends *any_compatible* to the end of the input array.
 
 ```sql title=Syntax
-array_dims ( array ) → string
+array_append ( array, any_compatible ) → array
 ```
 
 ```sql title=Example
-array_dims ( array[2,3,4] ) → [1:3]
+array_append (array[66], array[123]) → {66, 233}
```
Comment (xiangjinwu) [inline docs/sql/functions-operators/sql-function-array.md:16]: This example is wrong in 2 places:
* Input is `123` while output is `233`
* The 2nd arg should not be an array

**Review by xiangjinwu — COMMENTED** [review]
Two more things:
* `array_dims` was removed in this PR by accident
* `||` also supports `array_append` and `array_prepend`:
```
array[66] || 123 -> {66, 123}
123 || array[66] -> {123, 66}
```
[More](https://www.postgresql.org/docs/15/arrays.html#:~:text=In%20simple%20cases%2C%20the%20concatenation%20operator%20discussed%20above%20is%20preferred%20over%20direct%20use%20of%20these%20functions.%20However%2C%20because%20the%20concatenation%20operator%20is%20overloaded%20to%20serve%20all%20three%20cases%2C%20there%20are%20situations%20where%20use%20of%20one%20of%20the%20functions%20is%20helpful%20to%20avoid%20ambiguity.) in PostgreSQL for `||` vs functions.

**Pull request merged**